system/CLI/InputOutput.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of CodeIgniter 4 framework.
+ *
+ * (c) CodeIgniter Foundation <admin@codeigniter.com>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace CodeIgniter\CLI;
+
+/**
+ * Input and Output for CLI.
+ */
+class InputOutput
+{
+    /**
+     * Is the readline library on the system?
+     */
+    private bool $readlineSupport;
+
+    public function __construct()
+    {
+        // Readline is an extension for PHP that makes interactivity with PHP
+        // much more bash-like.
+        // http://www.php.net/manual/en/readline.installation.php
+        $this->readlineSupport = extension_loaded('readline');
+    }
+
+    /**
+     * Get input from the shell, using readline or the standard STDIN
+     *
+     * Named options must be in the following formats:
+     * php index.php user -v --v -name=John --name=John
+     *
+     * @param string|null $prefix You may specify a string with which to prompt the user.
+     */
+    public function input(?string $prefix = null): string
+    {
+        // readline() can't be tested.
+        if ($this->readlineSupport && ENVIRONMENT !== 'testing') {
+            return readline($prefix); // @codeCoverageIgnore
+        }
+
+        echo $prefix;
+
+        $input = fgets(fopen('php://stdin', 'rb'));
+
+        if ($input === false) {
+            $input = '';
+        }
+
+        return $input;
```

Comment:
Are we forbidding the Elvis operator?
```suggestion
        return fgets(fopen('php://stdin', 'rb')) ?: '';
```

Reply:
> Disallow short ternary operator (?:) - implies weak comparison, it's recommended to use null coalesce operator (??) or ternary operator with strict condition.
https://github.com/phpstan/phpstan-strict-rules#extra-strict-and-opinionated-rules-for-phpstan

Reply:
Humph.